pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -123,11 +162,15 @@ func Resolve(tpl integration.Config, svc listeners.Service) (integration.Config,
 // substituteTemplateVariables replaces %%VARIABLES%% in the config init,
 // instances, and logs config.
 // When there is an error, it stops processing.
-func substituteTemplateVariables(ctx context.Context, config *integration.Config, svc listeners.Service) error {
+func substituteTemplateVariables(ctx context.Context, config *integration.Config, svc listeners.Service, postProcessor func(interface{}) error) error {
```

Comment:
Godoc for `postProcessor` would be good

Reply:
I’ve just enriched the godoc to better describe the purpose of the `postProcessor` function parameter.